common/lib/dependabot/git_metadata_fetcher.rb

```diff
@@ -109,15 +115,20 @@ def fetch_raw_upload_pack_with_git_for(uri)
       service_pack_uri += ".git" unless service_pack_uri.end_with?(".git")
 
       env = { "PATH" => ENV.fetch("PATH", nil) }
-      command = "git ls-remote #{service_pack_uri}"
-      command = SharedHelpers.escape_command(command)
 
+      command = SharedHelpers.escape_command("git --version") # Is git present?
       stdout, stderr, process = Open3.capture3(env, command)
-      # package the command response like a HTTP response so error handling
-      # remains unchanged
+      if process.success?
+        command = "git ls-remote #{service_pack_uri}"
+        command = SharedHelpers.escape_command(command)
+        stdout, stderr, process = Open3.capture3(env, command)
+      end
+
+      # package the command response like a HTTP response so error handling remains unchanged
       if process.success?
         OpenStruct.new(body: stdout, status: 200)
       else
+        # this picks up the response from either `git --version` or `git ls-remote`, whichever fails first
```

Comment:
the fact that I needed to add a code comment may indicate this is overly clever, not sure...